packages/core/core/src/ResolverRunner.js

```diff
@@ -28,6 +29,35 @@ export default class ResolverRunner {
     this.pluginOptions = new PluginOptions(this.options);
   }
 
+  async getThrowableDiagnostic(dependency: Dependency, message: string) {
+    let diagnostic: Diagnostic = {
+      message,
+      origin: '@parcel/resolver-default'
+    };
+
+    if (dependency.loc && dependency.sourcePath) {
+      diagnostic.filePath = dependency.sourcePath;
+      diagnostic.codeFrame = {
+        code: await this.options.inputFS.readFile(
```

Comment:
How would I go about getting the asset source in the resolver?

Reading it from fileSystem would result in incorrect codeFrames... It has to be a snapshot from the moment the location data was extracted. Which is not possible atm afaik...

Reply:
Hmm, well ideally it *would* be based on the original source code. That's what the user wrote, so it makes sense to show them errors based on that (not generated code). The `loc` should be based on the original source code too. If only AST transforms happen (in the case of babel), then this should be true, otherwise we should be passing an input source map through as part of the transformation (e.g. TSC -> JS - doesn't seem to be the case currently though 😞).

Reply:
So there's no way we could fix this? I tried it on Babel btw so it should've returned the correct location? Which it didn't

Reply:
Hmm I’m confused. Why would the error location be incorrect?

Reply:
This is what I get when using the dependency location on the source

![Screenshot 2019-11-17 at 21 57 49](https://user-images.githubusercontent.com/2175521/69013920-50da3700-0985-11ea-974a-39616d1041ac.png)

Reply:
Nevermind, it was 0-based, works now. I thought every location in Babel was 1-based as there codeframes where as well...

![Screenshot 2019-11-17 at 22 03 50](https://user-images.githubusercontent.com/2175521/69014011-58e6a680-0986-11ea-88a6-31cae8173e18.png)

Reply:
https://github.com/babel/babel/blob/master/packages/babel-parser/ast/spec.md#node-objects:
> Each Position object consists of a line number (1-indexed) and a column number (0-indexed):

Reply:
Ow got it will not add +1 on columns